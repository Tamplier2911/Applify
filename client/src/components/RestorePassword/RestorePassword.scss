.restore-password {
  display: grid;
  @media only screen and (max-width: 450px) {
    margin-bottom: 5rem;
  }
}
