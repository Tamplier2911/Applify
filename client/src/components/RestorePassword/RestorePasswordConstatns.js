const restorePasswordData = {
  eng: {
    restorePasswordTitle: "Restore password:",
  },
  rus: { restorePasswordTitle: "Восстановить пароль:" },
  ukr: { restorePasswordTitle: "Відновити пароль:" },
};

export default restorePasswordData;
