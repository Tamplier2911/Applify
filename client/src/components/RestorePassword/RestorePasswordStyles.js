import styled from "styled-components";

export const RestorePasswordContainer = styled.div`
  display: grid;
  @media only screen and (max-width: 450px) {
    margin-bottom: 5rem;
  }
`;
