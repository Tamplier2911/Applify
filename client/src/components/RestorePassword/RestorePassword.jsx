// import "./RestorePassword.scss";
import React from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";

// components
import FormHolder from "../FormHolder/FormHolder";
import RestorePasswordForm from "../RestorePasswordForm/RestorePasswordForm";

// JS Rendering CSS
import { RestorePasswordContainer } from "./RestorePasswordStyles";

// component constants
import restorePasswordData from "./RestorePasswordConstatns";

const RestorePassword = ({ lang }) => {
  const { restorePasswordTitle } = restorePasswordData[lang];

  return (
    <RestorePasswordContainer data-test="restore-password">
      <FormHolder
        data-test="restore-password-form-holder"
        type="restore"
        title={restorePasswordTitle}
      >
        <RestorePasswordForm data-test="restore-password-form" />
      </FormHolder>
    </RestorePasswordContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage,
});

export default connect(mapStateToProps)(RestorePassword);
