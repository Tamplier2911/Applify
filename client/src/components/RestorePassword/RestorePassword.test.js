import React from "react";
import toJson from "enzyme-to-json";

// component
import RestorePassword from "./RestorePassword";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<RestorePassword />", () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<RestorePassword store={store} />);
  });

  it("renders without an error", () => {
    const restorePassword = findByTestAttr(wrapper, "restore-password");
    expect(restorePassword).toHaveLength(1);
  });

  it("renders children - form holder", () => {
    const restorePasswordFormHolder = findByTestAttr(
      wrapper,
      "restore-password-form-holder"
    );
    expect(restorePasswordFormHolder).toHaveLength(1);
  });

  it("renders children - form", () => {
    const restorePasswordForm = findByTestAttr(
      wrapper,
      "restore-password-form"
    );
    expect(restorePasswordForm).toHaveLength(1);
  });

  it("matches snapshot", () => {
    const restorePassword = findByTestAttr(wrapper, "restore-password");
    expect(toJson(restorePassword)).toMatchSnapshot();
  });
});
