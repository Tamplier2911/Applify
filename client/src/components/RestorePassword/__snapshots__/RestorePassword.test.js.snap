// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestorePassword /> matches snapshot 1`] = `
<styled.div
  data-test="restore-password"
>
  <FormHolder
    data-test="restore-password-form-holder"
    title="Restore password:"
    type="restore"
  >
    <withRouter(Connect(RestorePasswordForm))
      data-test="restore-password-form"
    />
  </FormHolder>
</styled.div>
`;
